crates/wasi-http/src/http_impl.rs

```diff
@@ -183,7 +183,8 @@ impl WasiHttp {
         let body = Full::<Bytes>::new(
             self.streams
                 .get(&request.body)
-                .unwrap_or(&Bytes::new())
+                .unwrap_or(&Stream::new())
```

Comment:
If we had `Default` implemented, this could have been `unwrap_or_default`

Reply:
Unfortunately, `HashMap::get(...)` returns a reference rather than a value, and there's no default for a reference. I could implement the `Default` trait for `&Stream` references, but I'm not sure if that's what is best. Let me know.

Reply:
That makes sense, I think this is fine.
I am making a few changes here to allow for the host to provide a custom connect implementation, so I'll take a deeper look at this as well next week